scripts/variantstore/wdl/GvsCreateVATFromAnnotations.wdl

```diff
@@ -0,0 +1,549 @@
+        String table_suffix
+        String? service_account_json_path
+    }
+
+    Array[String] contig_array = ["chr1", "chr2", "chr3", "chr4", "chr5", "chr6", "chr7", "chr8", "chr9", "chr10", "chr11", "chr12", "chr13", "chr14", "chr15", "chr16", "chr17", "chr18", "chr19", "chr20", "chr21", "chr22", "chrX", "chrY", "chrM"]
+    File reference = "gs://gcp-public-data--broad-references/hg38/v0/Homo_sapiens_assembly38.fasta"
+
+    call GetAnnotations {
+        input:
+            service_account_json_path = service_account_json_path,
+            inputFileofFileNames = inputFileofFileNames
+    }
+
+    ## Scatter across the shards from the GVS jointVCF
+    scatter(i in range(length(GetAnnotations.input_jsons)) ) {
+        ## Parse the annotations
+        call PrepAnnotationJson {
+          input:
+            annotation_json = GetAnnotations.input_jsons[i],
+            output_file_suffix = basename(GetAnnotations.input_jsons[i], "_annotated.json.gz") + ".json.gz",
+            output_path = output_path,
+            service_account_json_path = service_account_json_path
+        }
+    }
+
+    call BigQueryLoadJson {
+       input:
+         nirvana_schema = vat_schema_json_file,
+         vt_schema = variant_transcript_schema_json_file,
+         genes_schema = genes_schema_json_file,
+         project_id = project_id,
+         dataset_name = dataset_name,
+         output_path = output_path,
+         table_suffix = table_suffix,
+         service_account_json_path = service_account_json_path,
+         prep_jsons_done = PrepAnnotationJson.done
+  }
+
+    scatter(i in range(length(contig_array)) ) {
+      call BigQueryExportVat {
+        input:
+            contig = contig_array[i],
+            project_id = project_id,
+            dataset_name = dataset_name,
+            output_path = output_path,
+            table_suffix = table_suffix,
+            service_account_json_path = service_account_json_path,
+            load_jsons_done = BigQueryLoadJson.done
+      }
+    }
+}
+
+################################################################################
+
+task GetAnnotations {
+    input {
+        String? service_account_json_path
+        File inputFileofFileNames
+    }
+    parameter_meta {
+        inputFileofFileNames: {
+          localization_optional: true
+        }
+    }
+    String has_service_account_file = if (defined(service_account_json_path)) then 'true' else 'false'
+    String updated_input_files = if (defined(service_account_json_path)) then basename(inputFileofFileNames) else inputFileofFileNames
+
+    command <<<
+        set -e
+
+        if [ ~{has_service_account_file} = 'true' ]; then
+          gsutil cp ~{service_account_json_path} local.service_account.json
+          export GOOGLE_APPLICATION_CREDENTIALS=local.service_account.json
+          gcloud auth activate-service-account --key-file=local.service_account.json
+
+          gsutil cp ~{inputFileofFileNames} ~{updated_input_files}
```

Comment:
Looks this line is meant to be outside the `if`? If `has_service_account_file` is not `true` this command block will do nothing and the output expression for `input_jsons` might not be happy about that. 🙂

Reply:
line 77 should save me in this scenario, but soon, no SA!